packages/atomic/src/components/search/atomic-generated-answer/atomic-generated-answer.tsx

```diff
@@ -0,0 +1,143 @@
+import {
+  SearchStatus,
+  SearchStatusState,
+  buildSearchStatus,
+  buildGeneratedAnswer,
+  GeneratedAnswer,
+  GeneratedAnswerState,
+} from '@coveo/headless';
+import {Component, h, State} from '@stencil/core';
+import {
+  BindStateToController,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../utils/initialization-utils';
+import {Button} from '../../common/button';
+import {Bindings} from '../atomic-search-interface/atomic-search-interface';
+import {FeedbackButton} from './feedback-button';
+import {SourceCitations} from './source-citations';
+import {TypingLoader} from './typing-loader';
+
+/**
+ * @internal
+ */
+@Component({
+  tag: 'atomic-generated-answer',
+  styleUrl: 'atomic-generated-answer.pcss',
+  shadow: true,
+})
+export class AtomicGeneratedAnswer implements InitializableComponent {
+  @InitializeBindings() public bindings!: Bindings;
+  public generatedAnswer!: GeneratedAnswer;
+  public searchStatus!: SearchStatus;
+
+  @BindStateToController('generatedAnswer')
+  @State()
+  private generatedAnswerState!: GeneratedAnswerState;
+
+  @BindStateToController('searchStatus')
+  @State()
+  private searchStatusState!: SearchStatusState;
+
+  @State()
+  public error!: Error;
+
+  @State()
+  hidden = true;
+
+  public initialize() {
+    this.generatedAnswer = buildGeneratedAnswer(this.bindings.engine);
+    this.searchStatus = buildSearchStatus(this.bindings.engine);
+  }
+
+  private get hasRetryableError() {
+    return (
+      !this.searchStatusState.hasError &&
+      this.generatedAnswerState.error?.isRetryable
+    );
+  }
+
+  private get hideComponent() {
+    const {isLoading, answer, citations} = this.generatedAnswerState;
+    return (
+      !(isLoading || answer !== undefined || citations.length) &&
+      !this.hasRetryableError
+    );
+  }
+
+  private renderContent() {
+    return (
+      <div part="generated-content">
+        <div class="flex items-center">
+          <div part="header-label" class="text-bg-blue">
+            {this.bindings.i18n.t('generated-answer-title')}
+          </div>
+          {!this.generatedAnswerState.error && (
+            <div class="feedback-buttons flex">
+              <FeedbackButton
+                title={this.bindings.i18n.t('this-answer-was-helpful')}
```

Comment:
This label shouldn't be needed if you copy `atomic-smart-snippet-feedback-banner`.

Reply:
The label names contain "smart-snippets" is it ok to rename if I'm reusing them?

Reply:
sure, we should keep supporting the old ones as a fallback though, and add a TODO to remove them in v3.

For now you can merge this though since there's not a lot of time. Just gotta make sure to fix this before removing the internal flag.